buddy_alloc.h

```diff
@@ -369,7 +412,7 @@ struct buddy *buddy_init(unsigned char *at, unsigned char *main, size_t memory_s
     if (at_alignment != 0) {
         return NULL;
     }
-    size_t main_alignment = ((uintptr_t) main) % __alignof__(BUDDY_ALLOC_ALIGN);
+    size_t main_alignment = ((uintptr_t) main) % buddy_alignof(BUDDY_ALLOC_ALIGN);
```

Comment:
This might be a bug of original implementation
From documentation of __alignof__
If the operand is a `lvalue', the
return value represents the required alignment of the underlying type,
not the actual alignment of the specified `lvalue'.

Reply:
Looks like you're right, thanks for finding and fixing that! :)

Reply:
I didn't fix that per se, I wrapped that with macro, that does the same thing as before.